### ISSUE TITLE
`building=toilets` should trigger bathroom-related quests

### ISSUE THREAD

**Issue by Tex2002ans** [streetcomplete/StreetComplete#5620]
I believe bathroom-related quests are only triggered on:

- `amenity=toilets`

but not:

- `building=toilets`

### How to Reproduce

In StreetComplete 57.3, I changed a generic `building=yes` using this:

1. Enable "Building" overlay.
2. Click on any "red color" building.
3. Choose "Civic Buildings".
4. Choose "Toilets".
5. Press checkmark in lower right corner.
  - Leads to `building=yes` -> `building=toilets`.

It did not trigger bathroom quests though.

### Expected Behavior

After step 5, toilet-related quests should appear, such as:

- "Is this place wheelchair accessible?"
- "Is the restroom at this place wheelchair accessible?"

### Versions affected

StreetComplete 57.3 on Android.

- - -

And I just double-checked. The instant I:

- Tagged the building using OpenStreetMap.org's method
- Refreshed StreetComplete 57.3

the "Is this place wheelchair accessible?" quest popped up.

- - -

**Side Note:** On OpenStreetMap.org, choosing "Restroom" created:

- `building=yes`
- `amenity=toilets`

where StreetComplete 57.3's steps 1–5 above shifted:

- `building=yes` -> `building=toilets`

On the [OSM wiki for "Tag:amenity=toilets"](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dtoilets), I see this explanation:

> Alternatively **draw an area over the toilet area** and tag it with `amenity=toilets`, and **if located in a dedicated building with `building=toilets`**.

Hmmm... According to taginfo's stats:

- ~24.5k = [`building=toilets`](https://taginfo.openstreetmap.org/tags/building=toilets#overview)
- ~91.7k = [`building=yes` + `amenity=toilets`](https://taginfo.openstreetmap.org/tags/amenity=toilets#combinations)

**Comment by matkoniecz** [comment]
That is because you may have abandoned toilet building, in which case you do not have active toilet there.


https://www.walesonline.co.uk/news/wales-news/public-toilets-sold-newport-council-20683059

![screen01](https://github.com/streetcomplete/StreetComplete/assets/899988/cf9aecd1-3f5a-4a18-b8d8-d5044168bd19)

Note that with things overlay you can add missing active toilets.

**Comment by mnalis** [comment]
That is normal, and unfortunately a source of common OSM newcomer confusion. 

See _[FAQ: Building Type Quest - Why don't I see the right types?](https://wiki.openstreetmap.org/wiki/StreetComplete/FAQ#Building_Type_Quest_-_Why_don't_I_see_the_right_types?)_ and https://github.com/streetcomplete/StreetComplete/issues/3868#issuecomment-1066209292. 

In short, [building=*](https://wiki.openstreetmap.org/wiki/Key:building) (i.e. the thing that `Building type quest` and `Building overlay`  in StreetComplete update) only defines ***architectural style of the building***. That `building=*` tag is very rarely of actual interest to the common user on the ground which tries to find some facility, and who thus  would be interested in `amenity=*` or `shop=*` or `heathcare=*`.

It is **separate** from the purpose that building is **currently being used for** (if any). For example, you can have `building=church` which is being used as `amenity=pub`, or `building=bunker` which is used as `amenity=nightclub`. Or `building=hangar` of which half is being used as `amenity=marketplace` and half as `amenity=food_court`. 

Of course, you can also have `building=church` which is being used as `amenity=place_of_worship`, and it does in fact happens quite often, but it is not given...

---

In short, if you want to map missing toilets using StreetComplete, you should use `Things Overlay` and choose `toilets` from there. Then it will get tagged with `amenity=toilets` and you'll get all the followup quests. 

_(`Building Overlay` is of much less practical use, so I personally would use that only when I've completed everything else and am bored...)_

**Comment by matkoniecz** [comment]
> (Building Overlay is of much less practical use, so I personally would use that only when I've completed everything else and am bored...)

Or if you want to get address/building level/roof type quests.

**Comment by Tex2002ans** [comment]
@mnalis Ahh fantastic. Thanks again for the helpful links/info. :)

- - -

So... the *most correct* way for [this type of toilet building](https://playgroundprofessionals.com/sites/playgroundprofessionals.com/files/media/IMG_20150701_104902407_HDR-2.jpg) would be:

- `building=toilets`
- `amenity=toilets`

- - -

> (`Building Overlay` is of much less practical use, so I personally would use that only when I've completed everything else and am bored...)

Yes... so far, I've only used it to see some "red buildings" in the area, then tag:

- shed
- garage
- barn
- roof / [picnic shelter](https://github.com/openstreetmap/iD/issues/5347)
   - A roof with "picnic tables / benches / trash cans" underneath it.
   - "pavilion" is what we call those in the US... but according to UK/OSM... `building=pavilion` is something completely different.

A few buildings were impossible to tell with the satellite view or were in heavy tree cover, but in-person, it was easy to see what kind it was. :)

(I caught a few gazebos too... or this weird two-story viewing station... or in one town, there's this custom three-story "lighthouse"-thing a person built!)

**Comment by matkoniecz** [comment]
> So... the most correct way for [this type of toilet building](https://playgroundprofessionals.com/sites/playgroundprofessionals.com/files/media/IMG_20150701_104902407_HDR-2.jpg) would be:

I would say that both are 100% correct, though second provide a bit more info. Though standalone toilets are rarely in repurposed buildings, so second tagging schema is also fine.

I would rate second as worth 1000 points and first as worth 1001 points :)

> And that's how I stumbled across this initial issue. I was at the park, and one of the "generic gray rectangles" from satellite view turned out to be public bathrooms. I tagged it, then expected all the toilet quests to pop up too! :)

Oh, that is not really clear. Maybe it would be nice to suggest also marking is as in-use toilet? But it is tricky as `amenity=toilets` can be also present as standalone node within building....

**Comment by mnalis** [comment]
> So... the most correct way for [this type of toilet building](https://playgroundprofessionals.com/sites/playgroundprofessionals.com/files/media/IMG_20150701_104902407_HDR-2.jpg) would be:
>    building=toilets + amenity=toilets

Yes.

Although note that some might micromap that, e.g. have a polygon mapped as `building=toilets`, but then have separate nodes tagged [amenity=toilets](https://wiki.openstreetmap.org/wiki/Tag:amenity=toilets)+`male=yes`+`female=no` on one side, and `amenity=toilets`+`male=no`+`female=yes`+`changing_table=yes` on the other side of that building, maybe with `amenity=drinking_water` (it I see that correctly on the picture) in the middle (and maybe other sorts of details too).

That might be considered even "more correct", although probably not too useful on such small buildings, but more common on bigger ones like on camping grounds etc. (where you don't want to have to walk all around long building to enter correct toilet). 

But really, even just adding only single element as `amenity=toilets` is overwhelmingly the biggest improvement; all the extra effort after that is just icing on the cake... If you can detail-map 10 toilets, or add 11 bare-bones toilets, I'd recommend the latter (even if both actions took the same time, which is unlikely).

**Comment by Tex2002ans** [comment]
Hmmm... I hit this again a few days ago.

I was at a park, tagged a generic "red building" as `building=toilets`, and could've really used those "bathroom quests" while I was right there!

(Instead, I took a photo and waited until I got home to enhance the information right on the building itself.)

- - -

> I would rate second as worth 1000 points and first as worth 1001 points :)

And thanks for this info. Ever since you taught that `building` type to me—every time I run across that kind in ID, I update it "for 1001 points"! :) (Same with `building=church`es! Very easy to verify+add as I'm adding websites / phone numbers to it.)

**Comment by matkoniecz** [comment]
> (Instead, I took a photo and waited until I got home to enhance the information right on the building itself.)

You can also switch to things overlay - and add toilet if missing directly in SC

**Comment by Tex2002ans** [comment]
Wouldn't Things drop a toilet Node on it instead?

Thanks for the info. I'll definitely have to do some testing next time it happens. :)

**Comment by matkoniecz** [comment]
> Wouldn't Things drop a toilet Node on it instead?

Yes, and it is entirely fine. Merging POI with building is not needed (though for standalone toilet building it is not wrong/problematic at all to merge them, with shops/hotels/museums it can cause problems).
